### PR TITLE
Public API: ClusterRepo CRD

### PR DESCRIPTION
**Pull request by rohitsakala**
- Migrate ClusterRepo CRD

### CODE REVIEW

File: pkg/apis/catalog.cattle.io/v1/types.go [hunk 1]
```diff
@@ -6,20 +6,34 @@ import (
...
 // +genclient
-// +kubebuilder:skipversion
 // +genclient:nonNamespaced
+// +kubebuilder:resource:scope=Cluster,path=clusterrepos
+// +kubebuilder:subresource:status
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// ClusterRepo represents a particular helm repository and also contains
+// details about its location and credentials to connect to it for fetching
+// charts hosted in that particular helm repository.
```
Comment (pjbgf) [inline pkg/apis/catalog.cattle.io/v1/types.go:16]: nit:
```suggestion
// ClusterRepo represents a particular Helm repository. It contains details
// about the chart location and the credentials needed for fetching charts 
// hosted in that particular Helm repository.
```

File: pkg/apis/catalog.cattle.io/v1/types.go [hunk 2]
```diff
@@ -6,20 +6,34 @@ import (
...
 type ClusterRepo struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              RepoSpec   `json:"spec"`
-	Status            RepoStatus `json:"status"`
+
+	// Contains details about the helm repository that needs to be used.
```
Comment (pjbgf) [inline pkg/apis/catalog.cattle.io/v1/types.go:21]: nit:
```suggestion
	// RepoSec contains details about the Helm repository that needs to be used.
```

File: pkg/apis/catalog.cattle.io/v1/types.go [hunk 3]
```diff
@@ -6,20 +6,34 @@ import (
...
+	// More info: kubectl explain clusterrepo.spec
+	Spec RepoSpec `json:"spec"`
+
+	// Contains details of the helm repository that is currently being used in the cluster.
```
Comment (pjbgf) [inline pkg/apis/catalog.cattle.io/v1/types.go:25]:
```suggestion
	// RepoStatus contains details of the Helm repository that is currently being used in the cluster.
```

File: pkg/apis/catalog.cattle.io/v1/types.go [hunk 4]
```diff
@@ -6,20 +6,34 @@ import (
...
+	// More info: kubectl explain clusterrepo.status
+	// +optional
+	Status RepoStatus `json:"status,omitempty"`
 }
 
-// SecretReference a reference to a secret object
+// SecretReference references to a secret object which contains the credentials.
```
Comment (pjbgf) [inline pkg/apis/catalog.cattle.io/v1/types.go:31]:
```suggestion
// SecretReference references a Secret object which contains the credentials.
```

File: pkg/apis/catalog.cattle.io/v1/types.go [hunk 5]
```diff
@@ -51,34 +66,37 @@ type RepoSpec struct {
...
-	// It is expected the secret be of type "kubernetes.io/basic-auth" or "kubernetes.io/tls" for Helm repos
-	// and "kubernetes.io/basic-auth" or "kubernetes.io/ssh-auth" for git repos.
-	// For a repo the Namespace file will be ignored
+	// ClientSecret is the client secret to be used when connecting to a helm repository.
+	// The expected secret type is "kubernetes.io/basic-auth" or "kubernetes.io/tls" for Helm repositories
+	// and "kubernetes.io/basic-auth" or "kubernetes.io/ssh-auth" for Github helm repositories.
 	ClientSecret *SecretReference `json:"clientSecret,omitempty"`
 
-	// BasicAuthSecretName is the client secret to be used to connect to the repo
+	// BasicAuthSecretName is the client secret to be used to connect to the helm repository.
```
Comment (pjbgf) [inline pkg/apis/catalog.cattle.io/v1/types.go:78]:
```suggestion
	// BasicAuthSecretName is the client secret to be used to connect to the Helm repository.
```

**Review by pjbgf — APPROVED** [review]
A few optional nits, but overall LGTM.

**Review by diogoasouza — APPROVED** [review]

**Pull request merged**